xenmgr/XenMgr/Connect/Xl.hs

```diff
@@ -69,17 +70,25 @@ import System.Process
 import System.Directory
 import XenMgr.Rpc
 import XenMgr.Db
+import XenMgr.Errors
 import qualified Data.Map as M
 
 type NotifyHandler = [String] -> Rpc ()
 type Params = [(String, String)]
 
+data XlExceptionClass = OOMException String | XlException String
+    deriving (Typeable)
+instance Exception XlExceptionClass
+
+instance Show XlExceptionClass where
+    show e@(XlException s) = show s
```

Comment:
This function definition is non-exhaustive and doesn't compile:

    | XenMgr/Connect/Xl.hs:84:4:
    |     Warning: Pattern match(es) are non-exhaustive
    |              In the definition of `show': Patterns not matched: OOMException _